docs/reference/changelog.rst

```diff
@@ -16,7 +16,7 @@ Changelog
 
   For a complete list of commits, check out the `X.Y.Z`_ release on GitHub.
 
-2.16.0 (2025-MM-DD)
+2.16.0 (2025-07-04)
```

Comment:
It is missing a `.. _release-2.16.0:` reference on top of that (and the definition of the reference at the bottom of the file)